File: 프로그래머스/lv3/43238. 입국심사/입국심사.js
function solution(n, times) {
    let answer = n * Math.max.apply(null, times);
    let left = 1;
    let right = n * Math.max.apply(null, times);
    
    while (left <= right) {
        let mid = Math.floor((left + right) / 2);
        let total = 0;
        for (let i = 0; i < times.length; i += 1) {
            total += Math.floor(mid / times[i])
        }
        if (total >= n) {
            right = mid - 1;
            answer = Math.min(answer, mid)
        } else {
            left = mid + 1
        }
    }
    
    return answer;
}

File: 프로그래머스/lv3/43238. 입국심사/입국심사.py
def solution(n, times):
    answer = max(times) * n
    
    left = min(times)
    right = max(times) * n
    
    while left <= right:
        total = 0
        mid = (left + right) // 2
        for i in range(len(times)):
            total += mid // times[i]
        if total >= n:
            right = mid - 1
            answer = min(answer, mid)
        else:
            left = mid + 1
    
    return answer